vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -468,15 +468,18 @@ def process_weights_after_loading(self, layer: Module) -> None:
                                                         requires_grad=False)
             return
 
-    def apply(self,
-              layer: torch.nn.Module,
-              x: torch.Tensor,
-              router_logits: torch.Tensor,
-              top_k: int,
-              renormalize: bool,
-              use_grouped_topk: bool,
-              topk_group: Optional[int] = None,
-              num_expert_group: Optional[int] = None) -> torch.Tensor:
+    def apply(
+        self,
+        layer: torch.nn.Module,
+        x: torch.Tensor,
+        router_logits: torch.Tensor,
+        top_k: int,
+        renormalize: bool,
+        use_grouped_topk: bool,
+        topk_group: Optional[int] = None,
+        num_expert_group: Optional[int] = None,
+        custom_routing_function: Optional[Callable] = None,
```

Comment:
To fix the the error in the `Weight Loading Multiple GPU Test` you'll have to add the `custom_routing_function` to the `apply` method in the `CompressedTensorsMoEMethod`:

https://github.com/vllm-project/vllm/blob/ef9baee3c52f719df64a646db72b6c4ede8a29a0/vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py#L259

Reply:
Hi @dsikka , I've modified this interface and all tests are passed.